app/upgrades/v14/upgrade_test.go

```diff
@@ -0,0 +1,66 @@
+package v14_test
+
+import (
+	"reflect"
+	"testing"
+
+	gamm "github.com/osmosis-labs/osmosis/v13/x/gamm/keeper"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/osmosis-labs/osmosis/v13/app/apptesting"
+	v14 "github.com/osmosis-labs/osmosis/v13/app/upgrades/v14"
+)
+
+type UpgradeTestSuite struct {
+	apptesting.KeeperTestHelper
+}
+
+func (suite *UpgradeTestSuite) SetupTest() {
+	suite.Setup()
+}
+
+func TestUpgradeTestSuite(t *testing.T) {
+	suite.Run(t, new(UpgradeTestSuite))
+}
+
+func (suite *UpgradeTestSuite) TestMigrateNextPoolIdAndCreatePool() {
+	suite.SetupTest() // reset
+
+	const (
+		expectedNextPoolId uint64 = 3
+	)
+
+	var (
+		gammKeeperType = reflect.TypeOf(&gamm.Keeper{})
+	)
+
+	ctx := suite.Ctx
+	gammKeeper := suite.App.GAMMKeeper
+	swaprouterKeeper := suite.App.SwapRouterKeeper
+
+	// Set next pool id to given constant, because creating pools doesn't
+	// increment id on current version
+	gammKeeper.SetNextPoolId(ctx, expectedNextPoolId)
+	nextPoolId := gammKeeper.GetNextPoolId(ctx)
+	suite.Require().Equal(expectedNextPoolId, nextPoolId)
+
+	// system under test.
+	v14.MigrateNextPoolId(ctx, gammKeeper, swaprouterKeeper)
+
+	// validate swaprouter's next pool id.
+	actualNextPoolId := swaprouterKeeper.GetNextPoolId(ctx)
+	suite.Require().Equal(expectedNextPoolId, actualNextPoolId)
+
+	// create a pool after migration.
+	actualCreatedPoolId := suite.PrepareBalancerPool()
+	suite.Require().Equal(expectedNextPoolId, actualCreatedPoolId)
+
```

Comment:
There is a check no line 53. Are you referring to that?

Reply:
Ahh I see what you mean. I don't think this is necessary since this should be tested by e2e and `x/swaprouter` module.

The goals of the previous checks are the following:
- pool id is migrated to an index in swaprouter
- we can create a pool post-migration without error

The requirement of the pool id being incremented is outside of the migrations scope